mflowgen/Tile_PE/constraints/constraints.tcl

```diff
@@ -11,52 +11,163 @@
-set_output_delay -clock ${clock_name} [expr ${dc_clock_period}*0.2] [get_ports reset_out]
-set_output_delay -clock ${clock_name} [expr ${dc_clock_period}*0.2] [get_ports stall_out]
-
-# Make all signals limit their fanout
-
-set_max_fanout 20 $dc_design_name
-
-# Make all signals meet good slew
-
-set_max_transition [expr 0.25*${dc_clock_period}] $dc_design_name
-
-# False paths
-
-set_false_path -to [get_ports hi]
-set_false_path -to [get_ports lo]
-
 if $::env(PWR_AWARE) {
-    source inputs/dc-dont-use-constraints.tcl
     source inputs/pe-constraints.tcl
-    set_dont_touch [get_cells -hierarchical *u_mux_logic*]
 } 
+
+# Case analysis/scenarios
+
+set alu_path PE_inst0/WrappedPE_inst0\$PE_inst0/ALU_inst0\$ALU_comb_inst0
+set fp_add_path $alu_path/magma_BFloat_16_add_inst0
+set fp_mul_path $alu_path/magma_BFloat_16_mul_inst0
+set mul_path $alu_path/magma_Bits_32_mul_inst0
+set add_path $alu_path/magma_Bits_17_add*
+
+set sb_path SB_ID0_5TRACKS_B16_PE
+set sb_reg_path $sb_path/REG_T*_B16
+
+# Constrain SB to ~100 ps
+set sb_delay 0.200
+
+########################################################################
+# FP ADD
+########################################################################
+create_scenario fp_add
+
+source inputs/common.tcl
+
+set_case_analysis 0 [get_pins $alu_path/alu[7]]
+set_case_analysis 0 [get_pins $alu_path/alu[6]]
+set_case_analysis 0 [get_pins $alu_path/alu[5]]
+set_case_analysis 1 [get_pins $alu_path/alu[4]]
+set_case_analysis 0 [get_pins $alu_path/alu[3]]
+set_case_analysis 1 [get_pins $alu_path/alu[2]]
+set_case_analysis 1 [get_pins $alu_path/alu[1]]
+set_case_analysis 0 [get_pins $alu_path/alu[0]]
+
+# set false path for combinational path from input->output
+set_false_path -from SB*_IN_* -to SB*_OUT_*
+set_false_path -from [all_inputs]
+set_false_path -to [all_outputs]
+
+set_false_path -through [get_pins [list $fp_mul_path/*]]
+set_false_path -through [get_pins [list $mul_path/*]]
+set_false_path -through [get_pins [list $add_path/*]]
+
+set fp_add_min [expr 0.95+0.03]
+set fp_add_delay [expr $fp_add_min+0.17]
+set_max_delay $fp_add_delay -through [get_pins [list $fp_add_path/*]]
+
+########################################################################
+# FP MUL
+########################################################################
+create_scenario fp_mul
+
+source inputs/common.tcl
+
+set_case_analysis 0 [get_pins $alu_path/alu[7]]
+set_case_analysis 0 [get_pins $alu_path/alu[6]]
+set_case_analysis 0 [get_pins $alu_path/alu[5]]
+set_case_analysis 1 [get_pins $alu_path/alu[4]]
+set_case_analysis 1 [get_pins $alu_path/alu[3]]
+set_case_analysis 0 [get_pins $alu_path/alu[2]]
+set_case_analysis 0 [get_pins $alu_path/alu[1]]
+set_case_analysis 1 [get_pins $alu_path/alu[0]]
+
+# set false path for combinational path from input->output
+set_false_path -from SB*_IN_* -to SB*_OUT_*
+set_false_path -from [all_inputs]
+set_false_path -to [all_outputs]
```

Comment:
Does this mean that for fp_mul, it will assume that tile inputs AND outputs are registered?

Reply:
We should make sure that there aren't any important paths being declared false here

Reply:
Yes, I did make the assumption that the fp_mul/fp_add tile inputs and outputs are registered.

Reply:
This scenario is no longer active though.